geomesa-core/src/main/scala/org/locationtech/geomesa/core/index/AttributeIdxStrategy.scala

```diff
@@ -74,43 +71,66 @@ trait AttributeIdxStrategy extends Strategy with Logging {
     output(s"Temporal filters: $temporalFilters")
     output(s"Other filters: $nonSTFilters")
 
-    val oFilter: Option[Filter] = filterListAsAnd(geomFilters ++ temporalFilters)
-    val oNonStFilters: Option[Filter] = nonSTFilters.map(_ => recomposeAnd(nonSTFilters)).headOption
+    val stFilter: Option[Filter] = filterListAsAnd(geomFilters ++ temporalFilters)
+    val ecqlFilter: Option[Filter] = nonSTFilters.map(_ => recomposeAnd(nonSTFilters)).headOption
```

Comment:
Good question... I think it's a semi-convoluted way of ensuring that the nonSTFilters isn't empty (as that would throw an exception in recomposeAnd). I'll change to an if.